client/rest-high-level/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -69,8 +71,21 @@ public String toString() {
                 '}';
     }
 
-    static Request ping() {
-        return new Request("HEAD", "/", Collections.emptyMap(), null);
+    // For developers please add your new methods in the alphabetical order
+
+    static Request delete(DeleteRequest deleteRequest) {
+        String endpoint = endpoint(deleteRequest.index(), deleteRequest.type(), deleteRequest.id());
+
+        Params parameters = Params.builder();
+        parameters.withRouting(deleteRequest.routing());
+        parameters.withParent(deleteRequest.parent());
+        parameters.withTimeout(deleteRequest.timeout());
+        parameters.withVersion(deleteRequest.version());
+        parameters.withVersionType(deleteRequest.versionType());
+        parameters.withRefreshPolicy(deleteRequest.getRefreshPolicy());
+        parameters.withWaitForActiveShards(deleteRequest.waitForActiveShards());
+
+        return new Request(HttpDelete.METHOD_NAME, endpoint, parameters.getParams(), null);
```

Comment:
yes something along those lines. we have existing tests for this in RequestTests, see e.g. testIndex